src/services/redisSessionStorage.ts

```diff
@@ -48,19 +52,64 @@ export default class RedisSessionStorage implements ISessionStorage {
       );
     });
 
-    const [setSessionTokenResult, setWalletTokenResult] = await Promise.all([
+    const newSessionInfo: SessionInfo = {
+      sessionToken: user.session_token,
+      timestamp: new Date()
+    };
+    const setUserTokenInfo = new Promise<Either<Error, boolean>>(resolve => {
+      this.redisClient.set(
+        `${userKeyPrefix}${user.fiscal_code}-${sessionKeyPrefix}${
+          user.session_token
+        }`,
+        JSON.stringify(newSessionInfo),
+        "EX",
+        this.tokenDurationSecs,
+        (err, response) => resolve(this.singleStringReply(err, response))
+      );
+    });
+    const updateUserTokensSet = new Promise<Either<Error, boolean>>(resolve => {
+      this.redisClient.sadd(
+        `${userKeyPrefix}${user.fiscal_code}`,
+        `${userKeyPrefix}${user.fiscal_code}-${sessionKeyPrefix}${
+          user.session_token
+        }`,
+        (err, response) => resolve(this.integerReply(err, response))
+      );
+    });
+    const setPromisesResult = await Promise.all([
       setSessionToken,
-      setWalletToken
+      setWalletToken,
+      setUserTokenInfo,
+      updateUserTokensSet
     ]);
-
-    if (isLeft(setSessionTokenResult) || isLeft(setWalletTokenResult)) {
-      return left<Error, boolean>(new Error("Error setting the token"));
-    }
-
-    if (!setSessionTokenResult.value || !setWalletTokenResult.value) {
-      return left<Error, boolean>(new Error("Error setting the token"));
+    if (setPromisesResult.some(_ => isLeft(_) || !_.value)) {
+      return left<Error, boolean>(
+        new Error(
+          setPromisesResult
+            .map((_, index) => {
+              if (isLeft(_)) {
+                return _.value.message;
+              }
+              if (!_.value) {
+                switch (index) {
```

Comment:
this is very fragile, reordering the item in the code above will break this code below - instead, you can set the error message when each promise gets rejected - thus associating the error message to each promise.

Reply:
fixed in 98640ab